src/mongo/delegates/SearchIndexer.class.php

```diff
@@ -0,0 +1,293 @@
+        $resourceAndType->timeout($this->config->getMongoCursorTimeout());
+        foreach ($resourceAndType as $rt)
+        {
+            if (array_key_exists("rdf:type",$rt))
+            {
+
+                $rdfTypes = array();
+
+                if (array_key_exists('u',$rt["rdf:type"]))
+                {
+                    $rdfTypes[] = $rt["rdf:type"]['u'];
+                }
+                else
+                {
+                    // an array of types
+                    foreach ($rt["rdf:type"] as $type)
+                    {
+                        $rdfTypes[] = $type['u'];
+                    }
+                }
+
+                $docs = $searchDocGenerator->generateSearchDocumentsBasedOnRdfTypes($rdfTypes, $resourceUri, $context);
+                foreach($docs as $d){
+                    $documentsToIndex[] = $d;
+                }
+
+            }
+        }
+
+        foreach($documentsToIndex as $document) {
+            if(!empty($document)) $searchProvider->indexDocument($document);
+        }
+    }
+
+    /**
+     * @param string $searchDocumentType
+     * @param string|null $resourceUri
+     * @param string|null $context
+     * @param string|null $queueName
+     */
+    public function generateSearchDocuments($searchDocumentType, $resourceUri=null, $context=null, $queueName=null)
+    {
+        $t = new \Tripod\Timer();
+        $t->start();
+        // default the context
+        $contextAlias = $this->getContextAlias($context);
+        $spec = \Tripod\Mongo\Config::getInstance()->getSearchDocumentSpecification($this->getStoreName(), $searchDocumentType);
+        
+        if($resourceUri)
+        {            
+            $this->generateAndIndexSearchDocuments($resourceUri, $contextAlias, $spec['from'], $searchDocumentType);
+            return;
+        }
+
+
+        // default collection
+        $from = (isset($spec["from"])) ? $spec["from"] : $this->podName;
+
+        $types = array();
+        if (is_array($spec["type"]))
+        {
+            foreach ($spec["type"] as $type)
+            {
+                $types[] = array("rdf:type.u"=>$this->labeller->qname_to_alias($type));
+                $types[] = array("rdf:type.u"=>$this->labeller->uri_to_alias($type));
+            }
+        }
+        else
+        {
+            $types[] = array("rdf:type.u"=>$this->labeller->qname_to_alias($spec["type"]));
+            $types[] = array("rdf:type.u"=>$this->labeller->uri_to_alias($spec["type"]));
+        }
+        $filter = array('$or'=> $types);
+        if (isset($resource))
+        {
+            $filter["_id"] = array(_ID_RESOURCE=>$this->labeller->uri_to_alias($resource),_ID_CONTEXT=>$contextAlias);
+        }
+
+        $docs = $this->config->getCollectionForCBD($this->getStoreName(), $from)->find($filter);
+        $docs->timeout($this->config->getMongoCursorTimeout());
+        foreach ($docs as $doc)
+        {
+            if($queueName && !$resourceUri)
```

Comment:
Do the checks need to be `is_null` here if the default value is `null`? Probably not.. maybe..